pip/commands/list.py

```diff
@@ -193,17 +255,78 @@ def output_package(self, dist):
         else:
             return '%s (%s)' % (dist.project_name, dist.version)
 
-    def output_package_listing(self, installed_packages):
+    def output_package_listing(self, installed_packages, options=None):
         installed_packages = sorted(
             installed_packages,
             key=lambda dist: dist.project_name.lower(),
         )
-        for dist in installed_packages:
-            logger.info(self.output_package(dist))
+
+        if (hasattr(options, "columns") and
```

Comment:
This is very similar logic to what's in `run_outdated`. It could probably be refactored out.